src/mapgen/mapgen.cpp

```diff
@@ -971,6 +971,12 @@ GenerateNotifier::GenerateNotifier(u32 notify_on,
 }
 
 
+GenerateNotifier::~GenerateNotifier()
+{
+	m_notify_events.clear();
```

Comment:
not necessary, the `std::map` will be automatically destroyed and its contents freed

Reply:
Ok.

Reply:
Done.